src/vs/platform/mcp/node/npmPackageService.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { spawn } from 'child_process';
+import { createDecorator } from '../../instantiation/common/instantiation.js';
+import { ILogService } from '../../log/common/log.js';
+
+export const INpmPackageManagementService = createDecorator<INpmPackageManagementService>('INpmPackageManagementService');
+
+export interface INpmPackageManagementService {
+	readonly _serviceBrand: undefined;
+
+	getLatestPackageVersion(packageName: string): Promise<string>;
+	installPackage(packageName: string, version: string, location: string): Promise<void>;
+}
+
+export class NpmPackageService implements INpmPackageManagementService {
+	readonly _serviceBrand: undefined;
+
+	constructor(
+		@ILogService private readonly logService: ILogService,
+	) { }
+
+	async getLatestPackageVersion(packageName: string): Promise<string> {
+		return new Promise<string>((resolve, reject) => {
+			const npmProcess = spawn('npm', ['view', packageName, 'version'], {
```

Comment:
This is unfortunately going to be a bit of pain for users who have nvm/fnm/etc, especially on Windows where we don't do shell resolution. It may be better to run this in a terminal rather than just a subprocess so that you get the full environment.